script.js

```diff
@@ -9,6 +9,8 @@ extractHostname = (url) => {
   return hostname;
 }
 
+isFromLocalMachine = (origin) => (origin === "localhost" || origin === "127.0.0.1");
```

Comment:
I think it will match more cases using a regexp like:
`origin.match(/localhost|[0-9]{2,3}\.[0-9]{2,3}\.[0-9]{2,3}\.[0-9]{2,3}|::1|\.local|^$/gi)`

Reply:
@cedk This regex don't even match `127.0.0.1`. According to [this article on IP classes](https://www.meridianoutpost.com/resources/articles/IP-classes.php) I don't know if a regex would be powerful enough to determine a private IP address (and what about private networks extended on multiple places ?) Maybe an allow-list would better fit ?